### PR TITLE
[feature-pose] Declaration of template instanciation for FeaturePose::CLASS_NAME

### LINKED ISSUE #150: Clang : error on explicit specialization of 'CLASS_NAME'
On clang, when calling the macro DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN in feature-pose.cpp: 
```bash
sot_ws/src/sot/sot-core/src/feature/feature-pose.cpp:45:13: error: explicit specialization of 'CLASS_NAME' after instantiation
template <> DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN(FeaturePose_t, "FeaturePose");
             ^
/usr/local/include/dynamic-graph/factory.h:28:32: note: expanded from macro 'DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN'
   const std::string CLASSTYPE::CLASS_NAME = CLASSNAME;                         \
                                ^
sot_ws/src/sot/sot-core/include/sot/core/feature-pose.hh:172:39: note: explicit instantiation first required here
 extern template class SOT_CORE_DLLAPI FeaturePose<R3xSO3Representation>;
                                       ^
sot_ws/src/sot/sot-core/src/feature/feature-pose.cpp:47:1: error: explicit specialization of 'CLASS_NAME' after instantiation
 DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN(FeaturePoseSE3_t, "FeaturePoseSE3");
 ^
 /usr/local/include/dynamic-graph/factory.h:28:32: note: expanded from macro 'DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN'
   const std::string CLASSTYPE::CLASS_NAME = CLASSNAME;                         \
                                ^
sot_ws/src/sot/sot-core/include/sot/core/feature-pose.hh:171:39: note: explicit instantiation first required here
 extern template class SOT_CORE_DLLAPI FeaturePose<SE3Representation>;
```
clang version:
```
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
```

### PR DESCRIPTION
**Pull request by olivier-stasse**
Fix #150

### CODE REVIEW

File: include/sot/core/robot-utils.hh [hunk 1]
```diff
@@ -11,6 +11,28 @@
...
+#include <boost/property_tree/ptree.hpp>
+
+#ifdef BOOST_MPL_LIMIT_VECTOR_SIZE_PUSH
+#pragma pop_macro("BOOST_MPL_LIMIT_VECTOR_SIZE")
+#endif
+
+#ifdef BOOST_MPL_LIMIT_LIST_SIZE_PUSH
+#pragma pop_macro("BOOST_MPL_LIMIT_LIST_SIZE")
+#endif
+
```
Comment (jmirabel) [inline include/sot/core/robot-utils.hh:35]: - How reasonable is this ?
- Why is is required ?
- is this C++98 compatible ?

Comment (olivier-stasse) [reply include/sot/core/robot-utils.hh:35]: This is a mistake this was not intend to be submitted on this pull request.

I will push it on my private repository on a different branch.

Comment (olivier-stasse) [reply include/sot/core/robot-utils.hh]: This is fixed normally now

File: include/sot/core/feature-pose.hh [hunk 2]
```diff
@@ -167,6 +167,10 @@ template <typename T>
 Vector6d convertVelocity(const MatrixHomogeneous &M,
                          const MatrixHomogeneous &Mdes,
                          const Vector &faNufafbDes);
+
+template<> const std::string FeaturePose<SE3Representation>::CLASS_NAME;
+template<> const std::string FeaturePose<R3xSO3Representation>::CLASS_NAME;
```
Comment (jmirabel) [inline include/sot/core/feature-pose.hh:172]: I doubt this works for C++ 98 so it should be moved into the `#if` below.

**Comment by olivier-stasse** [comment]
fixed by 471c9337dc3c

**Pull request closed without merging**